NMS-14850: Review enlinkd documentation (#5499)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/topology/enlinkd/layer-2/cdp-discovery.adoc

= Cisco Discovery Protocol

The https://learningnetwork.cisco.com/s/article/cisco-discovery-protocol-cdp-x[Cisco Discovery Protocol (CDP)] is a proprietary link layer protocol developed by Cisco.
Network devices use the protocol to advertise their identities, capabilities, and neighbors.

CDP functions similarly to several proprietary protocols, such as the Link Layer Discovery Protocol (LLDP), Extreme Discovery Protocol, Foundry Discovery Protocol (FDP), Nortel Discovery Protocol (also known as SONMP), and Microsoft's Link Layer Topology Discovery (LLTD) protocol.

> CDP uses information provided by the http://www.circitor.fr/Mibs/Html/C/CISCO-CDP-MIB.php[CISCO-CDP-MIB] and http://www.circitor.fr/Mibs/Html/C/CISCO-VTP-MIB.php[CISCO-VTP-MIB] modules.

Reviewer: ```suggestion
CDP uses information the http://www.circitor.fr/Mibs/Html/C/CISCO-CDP-MIB.php[CISCO-CDP-MIB] and http://www.circitor.fr/Mibs/Html/C/CISCO-VTP-MIB.php[CISCO-VTP-MIB] modules provide.
```

> You can find generic information about the CDP process in the CDP Information box on any device's Node Detail page.

Reviewer: ```suggestion
You can find generic information about the CDP process in the CDP information box on any device's node detail page.
```

Author: Since this is referring to an actual box/page in the UI, should we keep the capitalization? Doing so would keep this in line with the style guide.

== CDP Topology Updater

The CDP Topology Updater consolidates data from the CDP Collector and provides it to the CDP OnmsTopology.
Only full bidirectional connections between two CDP-supported devices become edges in the topology.
For example, Node A and Node B are connected by a CDP edge if, and only if, there is a CDP MIB port connection from Node A to Node B, and vice versa.

NOTE: Only nodes with a running CDP process can be part of the link discovery process.
The data collected is similar to that displayed by running the `show cdp neighbor` command on a device's command line.
Linux and Windows servers do not have a CDP process running by default, and will not be part of the link discovery process.

Information gathered from supported OIDs is stored in the following database tables:

.Database tables related to CDP discovery
image::enlinkd/cdp-database.png[Network diagram depicting database tables used in CDP discovery]

== Supported OIDs

The following OIDs are supported for the discovery and construction of the CDP network topology.

> They are collected by the CDP Discovery Collector:

Reviewer: ```suggestion
The CDP Discovery Collector collects them:
```

This will typically be the value of the `ifName` object (for example, `Ethernet0`). +
A zero-length string indicates that no `port-ID` field (TLV) was reported in the most recent CDP message.
| .1.3.6.1.4.1.9.9.23.1.2.1.1.7

| cdpCachePlatform
| Device's hardware platform, as reported in the most recent CDP message.
A zero-length string indicates that no `platform` field (TLV) was reported in the most recent CDP message.
| .1.3.6.1.4.1.9.9.23.1.2.1.1.8

| cdpGlobalRun
| Status of the Cisco Discovery Protocol.
Entries in `cdpCacheTable` are deleted when CDP is disabled.
| .1.3.6.1.4.1.9.9.23.1.3.1.0

| cdpGlobalDeviceId
| Device ID advertised by this device.
Its format is characterized by the value of `cdpGlobalDeviceIdFormat`.
| .1.3.6.1.4.1.9.9.23.1.3.4.0

| cdpGlobalDeviceIdFormat
| Format of the `device-Id` that is contained in the corresponding instance of `cdpGlobalDeviceId`.
You can specify only the formats that the device is capable of using, as denoted in `cdpGlobalDeviceIdFormatCpb`. +
*serialNumber(1):* Indicates that the value of `cdpGlobalDeviceId` is an ASCII string that contains the device's serial number. +

> *macAddress(2):* Indicates that the value of `cdpGlobalDeviceId` is a layer 2 MAC address. +

Reviewer: ```suggestion
*macAddress(2):* Indicates that the value of `cdpGlobalDeviceId` is a Layer 2 MAC address. +
```
